### PR TITLE
manage cache control when downloading working copy

### PR DESCRIPTION
**Pull request by ymerghoub**

### CODE REVIEW

File: eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/export/ExcelGenerator.java [hunk 1]
```diff
@@ -294,6 +295,11 @@ public File generateXLSResponse(QueryResult queryResult, Locale locale, String b
                     data.add(targetPartLinksAsString);
                     break;
 
+                case QueryField.PART_MASTER_IS_STANDARD:
```
Comment (florentgarin) [inline eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/export/ExcelGenerator.java]: This change has nothing to do with the issue. Must be removed from the PR.

File: eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/DocumentBinaryResource.java [hunk 2]
```diff
@@ -211,6 +212,16 @@ public Response downloadDocumentFile(
...
         InputStream binaryContentInputStream = null;
 
+        DocumentIteration lastIteration = documentService.getDocumentRevision(new DocumentIterationKey(workspaceId, documentId, version, iteration).getDocumentRevision()).getLastIteration();
+
+        Set<BinaryResource> binaryresources = lastIteration.getAttachedFiles();
+
+        boolean isCheckedOut = documentService.getDocumentRevision(new DocumentIterationKey(workspaceId, documentId, version, iteration).getDocumentRevision()).isCheckedOut();
+
+        boolean isWorkingCopy = binaryresources.contains(binaryResource) && isCheckedOut;
+
```
Comment (florentgarin) [inline eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/DocumentBinaryResource.java]: Why pass through all the binaryresource? Should be better to just check if the iteration is the latest one

File: eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/DocumentBinaryResource.java [hunk 3]
```diff
@@ -211,6 +212,16 @@ public Response downloadDocumentFile(
...
+        boolean isCached = !isWorkingCopy;
```
Comment (florentgarin) [inline eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/DocumentBinaryResource.java]: misnamed. "toPutInCache" should be better

File: eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/util/BinaryResourceDownloadResponseBuilder.java [hunk 4]
```diff
@@ -123,20 +124,25 @@ public static Response downloadError(Exception e, String fullName) {
...
-        Calendar expirationDate = Calendar.getInstance();
-        expirationDate.add(Calendar.SECOND, CACHE_SECOND);
-
-        return response.cacheControl(cc)
-                .expires(expirationDate.getTime())
-                .lastModified(lastModified)
+    private static Response.ResponseBuilder applyCachePolicyToResponse(Response.ResponseBuilder response, EntityTag eTag, Date lastModified, boolean isCached) {
+
+        if (isCached == true) {
+            CacheControl cc = new CacheControl();
```
Comment (florentgarin) [inline eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/util/BinaryResourceDownloadResponseBuilder.java]: if(isCached) is suffisant

Comment (ymerghoub) [reply eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/util/BinaryResourceDownloadResponseBuilder.java]: new commit after your comments

**Pull request closed without merging**